src/huak/ops/activate.rs

```diff
@@ -1,10 +1,14 @@
-use crate::{errors::HuakResult, project::Project};
+use crate::{
+    errors::{HuakError, HuakResult},
+    project::Project,
+};
 
 pub fn activate_project_venv(project: &Project) -> HuakResult<()> {
-    let venv = project
-        .venv()
-        .as_ref()
-        .expect("`Project::from` creates venv if it doesn't exists.");
+    let venv = project.venv().as_ref().ok_or_else(|| {
+        HuakError::InternalError("Project's venv is `None`.".to_owned())
```

Comment:
We should have HuakError::VenvNotFound for this

Reply:
I got confused again about when venv is or is not there. Because `Project::from` always creates `Venv` struct, so the only way `.venv()` could be `None` here is if `Project` would have been created using `Project::new` (which currently is only used in `huak new` command). But `Project::from` doesn't create a venv on the filesystem. So we still need to later check it venv does _actually exist_, even if it is `Some`.

So to sum up, `VenvNotFound` was used (by me at least) for when the venv does not exist on the filesystem. `venv` being `None`, to me, is more of a project error. That's why I used different error here.
All of that could be fixed if `Venv` struct would guarantee that venv does exist on the filesystem (i.e. `Venv::new` always does `.create()`).

I can change it to `VenvNotFound`, doesn't matter tbf. But the bigger issue still persists.

*Sorry for a bit lengthy architecture roast, but I assume you understand that I just want better code (and you probably do too)* :upside_down_face:

Reply:
No worries about the length. I love it. The fact that there's mutual interest around improving the design is amazing. So don't hesitate.

With that said, I think you're right. There's definitely more variety here, but I think using `VenvNotFound` where it *can* fit is better than trying to slap together a bunch of different .expects and hoping to (1) maintain consistency when you want it and (2) keep future refactors easier to iterate on.

So I think we use `VenvNotFound` for now and mark this as a TODO as part of #128 refactors. This is the sort of stuff that I think will contribute to a better future design.